### PR TITLE
Join users and groups in collab sidebar

### PR DESCRIPTION
**Pull request by kulmann**
## Description
In the collaborators sidebar, the users and groups were two separate lists. This PR joins all collaborators into a single list. In oc10 the collaborators were sorted by ids (ASC), now they are sorted by displayName (lower case, ascending), on equal names, groups are shown first.
This also fixes a visual offset to the left when displaying group shares.

## Related Issue

- Fixes https://github.com/owncloud/phoenix/issues/2900

## How Has This Been Tested?

Manual check in Chrome + Firefox and selenium tests with sharing context.

### oC10
Bildschirmfoto 2020-01-23 um 08 47 01

### Phoenix before change
Bildschirmfoto 2020-01-23 um 08 45 27

### Phoenix after change
Bildschirmfoto 2020-01-23 um 09 41 24

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: apps/files/src/components/FileSharingSidebar.vue [hunk 1]
```diff
@@ -125,15 +104,23 @@ export default {
...
-        return collaborator.info.share_type === '0' || collaborator.info.share_type === '6'
-      })
-    },
-    $_ocCollaborators_groups () {
-      return this.shares.filter(collaborator => {
-        return collaborator.info.share_type === '1'
-      })
+    $_ocCollaborators () {
+      return this.shares
+        .filter(collaborator => this.$_ocCollaborators_isUser(collaborator) || this.$_ocCollaborators_isGroup(collaborator))
```
Comment (PVince81) [inline apps/files/src/components/FileSharingSidebar.vue]: At first I thought we could get rid of the filter but then remembered that maybe link shares are also present in this structure ?

File: apps/files/src/components/FileSharingSidebar.vue [hunk 2]
```diff
@@ -125,15 +104,23 @@ export default {
...
+          const name2 = c2.displayName.toLowerCase().trim()
+          // sorting priority 1: display name (lower case, ascending), 2: share type (groups first), 3: id (ascending)
+          if (name1 === name2) {
+            if (this.$_ocCollaborators_isGroup(c1) === this.$_ocCollaborators_isGroup(c2)) {
+              return parseInt(c1.info.id, 10) < parseInt(c2.info.id, 10) ? -1 : 1
+            } else {
+              return this.$_ocCollaborators_isGroup(c1) ? -1 : 1
+            }
+          } else {
+            return name1 < name2 ? -1 : 1
```
Comment (PVince81) [inline apps/files/src/components/FileSharingSidebar.vue]: let's use natural sort, see https://github.com/owncloud/phoenix/blob/master/apps/files/src/fileSortFunctions.js#L28, it is used also in the file list

File: apps/files/src/components/FileSharingSidebar.vue (hunk 1 above)
Comment (kulmann) [reply apps/files/src/components/FileSharingSidebar.vue]: Yes, link shares are present in the shares list.

File: apps/files/src/components/FileSharingSidebar.vue (hunk 2 above)
Comment (kulmann) [reply apps/files/src/components/FileSharingSidebar.vue]: Using natural sort on the string comparison now (last bit of the sorting). Moved natural sort into `helpers/textUtils.js` for that.

File: changelog/unreleased/2900 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Join users and groups into a single list in collaborators sidebar
```
Comment (PVince81) [inline changelog/unreleased/2900]: I think this is rather a `Change`, not an enhancement ?
see changelog/README for the possible values

Comment (kulmann) [reply changelog/unreleased/2900]: Agreed, I have edited the commit 👍

**Pull request merged**